packages/gatsby/src/schema/resolvers.ts

```diff
@@ -0,0 +1,509 @@
+    const fieldNodes = getFieldNodeByNameInSelectionSet(
+      selectionSet,
+      fieldName,
+      info
+    )
+
+    const returnType = getNullableType(info.returnType)
+
+    if (isObjectType(returnType) || isInterfaceType(returnType)) {
+      const field = returnType.getFields()[fieldName]
+      if (field) {
+        const fieldArg = field.args.find(arg => arg.name === `field`)
+        if (fieldArg) {
+          const fieldEnum = getNullableType(fieldArg.type)
+
+          if (fieldEnum && isEnumType(fieldEnum)) {
+            return fieldNodes.reduce(
+              (acc: Array<string>, fieldNode: FieldNode) => {
+                const fieldArg = fieldNode.arguments?.find(
+                  arg => arg.name.value === `field`
+                )
+                if (fieldArg && fieldArg.value.kind === Kind.ENUM) {
+                  const enumKey = fieldArg.value.value
+                  const enumValue = fieldEnum.getValue(enumKey)
+                  if (enumValue) {
+                    return [...acc, enumValue.value]
+                  }
+                }
+                return acc
+              },
+              []
+            )
+          }
+        }
+      }
+    }
+  }
+
+  return []
+}
+
+function getFieldNodeByNameInSelectionSet(
+  selectionSet: SelectionSetNode,
+  fieldName: string,
+  info: GraphQLResolveInfo
+): Array<FieldNode> {
+  return selectionSet.selections.reduce(
+    (acc: Array<FieldNode>, selection: SelectionNode) => {
+      if (selection.kind === Kind.FRAGMENT_SPREAD) {
+        const fragmentDef = info.fragments[selection.name.value]
+        if (fragmentDef) {
+          return [
+            ...acc,
+            ...getFieldNodeByNameInSelectionSet(
+              fragmentDef.selectionSet,
+              fieldName,
+              info
+            ),
+          ]
+        }
+      } else if (selection.kind === Kind.INLINE_FRAGMENT) {
+        return [
+          ...acc,
+          ...getFieldNodeByNameInSelectionSet(
+            selection.selectionSet,
+            fieldName,
+            info
+          ),
+        ]
+      } /* FIELD_NODE */ else {
+        if (selection.name.value === fieldName) {
+          return [...acc, selection]
+        }
+      }
+      return acc
+    },
+    []
+  )
+}
+
+export const defaultFieldResolver: GatsbyResolver<
+  any,
+  any
+> = function defaultFieldResolver(source, args, context, info) {
+  if (
+    (typeof source == `object` && source !== null) ||
+    typeof source === `function`
+  ) {
+    if (info.from) {
+      if (info.fromNode) {
+        const node = context.nodeModel.findRootNodeAncestor(source)
+        if (!node) return null
+        return getValueAt(node, info.from)
+      }
+      return getValueAt(source, info.from)
+    }
+    const property = source[info.fieldName]
+    if (typeof property === `function`) {
+      return source[info.fieldName](args, context, info)
+    }
+    return property
+  }
+
+  return undefined
```

Comment:
We used to return `null` here (and we still return null on line 461), so is there a reason for changing it to undefined? Not sure if it can break something though.